musmart/timing/tempo_modeling.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""This module implements various functions related to linear tempo modeling e.g., `tempo_slope`, `tempo_drift`"""
+
+from typing import Iterable
+
+import numpy as np
+
+
+__all__ = ["tempo_slope", "tempo_drift", "tempo_fluctuation"]
+__author__ = "Huw Cheston"
+
+
+def _fit_tempo_linregress(beats: Iterable):
+    """Fits linear regression of BPM measurements vs onset time"""
+
+    from scipy.stats import linregress
+    # Dependent variable: BPM measurements
+    y = np.array(60 / np.diff(beats))
```

Comment:
`np.diff(beats)` will give weird results if any beats are skipped, it would probably be wise to mention that missing values are not permitted in the docstring